### PR TITLE
312 - adding chip selector in card

### PR DESCRIPTION
**Pull request by VinicciusSantos**

### CODE REVIEW

File: projects/ion/src/lib/card/card.component.spec.ts [hunk 1]
```diff
@@ -121,4 +132,21 @@ describe('CardComponent', () => {
...
   });
+
+  it('should not render chips by default', async () => {
+    cardComponent.cardConfig.header.chips = undefined;
+    fixture.detectChanges();
+    const chipsContainer = screen.queryByTestId('chips-container');
+    expect(chipsContainer).toBeNull();
+  });
+
+  it('should render chips', async () => {
```
Comment (iurynogueira) [inline projects/ion/src/lib/card/card.component.spec.ts:143]: Check chips has displaying in screen by label, I suggest use of it.each(yourChipsArray) and expect element with chip label on screen.

**Review by iurynogueira — CHANGES_REQUESTED** [review]

File: projects/ion/src/lib/card/card.component.spec.ts [hunk 2]
```diff
@@ -55,12 +60,18 @@ class CardTestComponent implements AfterViewInit {
 }
 
 @NgModule({
-  imports: [CommonModule, ButtonModule],
```
[the remark this answers is not in the document]
Comment (VinicciusSantos) [reply projects/ion/src/lib/card/card.component.spec.ts:58]: to use the ButtonModule, I need to make a little change to it by adding the DropdownComponent in the exports:
image

by doing this i would also remove the components declarations in .mdx file.
Can i do this?

File: projects/ion/src/lib/card/card.component.ts [hunk 3]
```diff
@@ -98,7 +103,13 @@ export class CardIonComponent implements AfterViewInit, OnDestroy {
     }
   }
 
-  cardEvent(type: ButtonType): void {
-    this.events.emit({ buttonAction: type });
+  cardEvent(event: CardEvent): void {
+    this.events.emit(event);
+    if (event['chipSelected'])
```
Comment (iurynogueira) [inline projects/ion/src/lib/card/card.component.ts:108]:
```suggestion
    if (event.chipSelected)
```

File: projects/ion/src/lib/card/card.component.ts [hunk 4]
```diff
@@ -98,7 +103,13 @@ export class CardIonComponent implements AfterViewInit, OnDestroy {
...
+      this.indexOfChipSelected = event.chipSelected['index'];
```
Comment (iurynogueira) [inline projects/ion/src/lib/card/card.component.ts:109]:
```suggestion
      this.indexOfChipSelected = event.chipSelected.index;
```

**Review by iurynogueira — CHANGES_REQUESTED** [review]

**Review by iurynogueira — APPROVED** [review]

**Pull request merged**